js/map.mjs

```diff
@@ -44,9 +36,44 @@ let mapAdjust = {
 let screenOriginInMapCoords = new Point(0,0);
 let _iconH = 20;
 function iconH(){return _iconH;};
-let currentOverlay = "Locations"; // Locations, NirnRoute, Exploration.
+let currentOverlay = "Locations"; // Locations, NirnRoute
 let showTSP = false;
 
+let randomGateCount = 0;
+function updateRandomGateCount(Found){
+    if(Found){
+        randomGateCount++;
+    }
+    else{
+        randomGateCount--;
+    }
+
+    if(randomGateCount >= 40){
+        document.getElementById("random_Gate_Count").innerText = randomGateCount + "✔";
+        document.getElementById("random_Gate_Count").style = "color:green";
+    }
+    else{
+        document.getElementById("random_Gate_Count").innerText = randomGateCount;    
+        document.getElementById("random_Gate_Count").style = "color:black";
+    }
+}
+function initRandomGateCount(){
+    //Init randomGateCount. 
+        //Is there a better way to do this? Only issue I think this presents is if we change the location IDs.
```

Comment:
once random gates are in their own section:
```
let root = window.findOnTree(window.jsondata.location, x=>x.name == "Random Gates", y=>y.name == "Random Gates" || y.elements == null)
let completed = window.sumCompletionItems(root);
randomGateCount = completed[0];
```

Reply:
Implemented.

Sectioned the random gates like so
```
{"name":"Oblivion Gates", "elements":[
    Fixed/City gates...
    {"name":"Random Gates", "elements":[
        Randomd Gates...
    ]}
]},
```